### PR TITLE
Add mlperf bert model

### PR DESCRIPTION
**Pull request by wozeparrot**
Language models are so much easier to write than RNN-T :sweat_smile:

### TODO

- [x] Dataset downloader
- [x] Metrics
- [x] ~~Do a run~~ ~~Edit: initial run at 90.539% f1~~ Edit Again: now matching the requirement of 90.874%

### CODE REVIEW

File: datasets/squad.py [hunk 1]
```diff
@@ -0,0 +1,149 @@
+import json
+import pathlib
+from transformers import BertTokenizer
+import numpy as np
+from extra.utils import download_file
+
+BASEDIR = pathlib.Path(__file__).parent.parent / "datasets/squad"
+download_file("https://rajpurkar.github.io/SQuAD-explorer/dataset/dev-v1.1.json", BASEDIR / "dev-v1.1.json")
```
Comment (wozeparrot) [inline datasets/squad.py:8]: I'm not sure if this is the best place to put the "downloader". It feels kinda weird.

Comment (geohot) [reply datasets/squad.py:8]: I'd put it in iterate, it shouldn't trigger on import.

**Comment by wozeparrot** [comment]
This reaches the required f1 score of 90.874% now.

**Comment by geohot** [comment]
Cool, ready for me to test?

**Comment by wozeparrot** [comment]
yup

**Comment by geohot** [comment]
Is that "0.72 Mask-LM accuracy"? From https://mlcommons.org/en/training-normal-21/

**Comment by geohot** [comment]
FileNotFoundError: [Errno 2] No such file or directory: '/tmp/tmpp9zuhu2c' -> '/home/tiny/tinygrad/datasets/squad/dev-v1.1.json'

**Comment by wozeparrot** [comment]
This is on the inference benchmark here: https://mlcommons.org/en/inference-edge-30/

want me to switch to the training dataset instead? its the same model

**Comment by geohot** [comment]
Can we do both datasets? They should be small.

**Comment by wozeparrot** [comment]
it seems like the training validation eval and the inference benchmark are using a slightly different model, as the training is only on Bert Base without a linear layer after for question answering

**Comment by geohot** [comment]
Running now, but wow how long did this take to finish?!? Getting 700 ms / example with OPTLOCAL on 7900XTX

**Comment by wozeparrot** [comment]
It's not batched and the reference impl isn't batched either. The 700ms / example is interesting cuz it takes me about 700ms / example as well on a 6800s.

**Comment by geohot** [comment]
Oh, it's not GPU bound at all...can you JIT or are they all different sizes?

**Comment by wozeparrot** [comment]
it does seem like jit works. will push

**Comment by wozeparrot** [comment]
getting around ~300ms an example now

**Comment by geohot** [comment]
Oh hell yea, 51 ms per sample with the JIT!

**Comment by geohot** [comment]
Final error rate, 90.874%! The train error is an issue for the training bounty

Same payment method?

**Comment by wozeparrot** [comment]
yup

**Comment by geohot** [comment]
Sent

**Pull request merged**